pkg/apis/aws/cache/v1alpha1/replication_group_types.go

```diff
@@ -0,0 +1,373 @@
+
+	Spec   ReplicationGroupSpec   `json:"spec,omitempty"`
+	Status ReplicationGroupStatus `json:"status,omitempty"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ReplicationGroupList contains a list of ReplicationGroup
+type ReplicationGroupList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []ReplicationGroup `json:"items"`
+}
+
+// NewReplicationGroupSpec creates a new ReplicationGroupSpec
+// from the given properties map.
+func NewReplicationGroupSpec(properties map[string]string) *ReplicationGroupSpec {
+	spec := &ReplicationGroupSpec{
+		ReclaimPolicy: corev1alpha1.ReclaimRetain,
+
+		// Note that these keys should match the JSON tags of their respective
+		// ReplicationGroupSpec fields.
+		AuthToken:                  properties["authToken"],
+		CacheNodeType:              properties["cacheNodeType"],
+		CacheParameterGroupName:    properties["cacheParameterGroupName"],
+		CacheSubnetGroupName:       properties["cacheSubnetGroupName"],
+		EngineVersion:              properties["engineVersion"],
+		NotificationTopicARN:       properties["notificationTopicArn"],
+		PreferredMaintenanceWindow: properties["preferredMaintenanceWindow"],
+		SnapshotName:               properties["snapshotName"],
+		SnapshotWindow:             properties["snapshotWindow"],
+		CacheSecurityGroupNames:    parseSlice(properties["cacheSecurityGroupNames"]),
+		PreferredCacheClusterAZs:   parseSlice(properties["preferredCacheClusterAzs"]),
+		SecurityGroupIDs:           parseSlice(properties["securityGroupIds"]),
+
+		// TODO(negz): Support NodeGroupConfiguration? It's awkward to extract a
+		// slice of structs from a one dimensional map - we might use Terraform
+		// style nodeGroupConfiguration.0.slots style notation. Without
+		// NodeGroupConfiguration we cannot support SnapshotARNs.
+	}
+
+	if b, err := strconv.ParseBool(properties["atRestEncryptionEnabled"]); err == nil {
+		spec.AtRestEncryptionEnabled = b
+	}
+	if b, err := strconv.ParseBool(properties["automaticFailoverEnabled"]); err == nil {
+		spec.AutomaticFailoverEnabled = b
+	}
+	if b, err := strconv.ParseBool(properties["transitEncryptionEnabled"]); err == nil {
+		spec.TransitEncryptionEnabled = b
+	}
+	if i, err := strconv.Atoi(properties["numCacheClusters"]); err == nil {
+		spec.NumCacheClusters = i
+	}
+	if i, err := strconv.Atoi(properties["numNodeGroups"]); err == nil {
+		spec.NumNodeGroups = i
+	}
+	if i, err := strconv.Atoi(properties["port"]); err == nil {
+		spec.Port = i
+	}
+	if i, err := strconv.Atoi(properties["replicasPerNodeGroup"]); err == nil {
+		spec.ReplicasPerNodeGroup = i
+	}
+	if i, err := strconv.Atoi(properties["snapshotRetentionLimit"]); err == nil {
+		spec.SnapshotRetentionLimit = i
+	}
+
+	return spec
+}
+
+// parseSlice parses a string of comma separated strings, for example
+// "value1, value2", into a string slice.
+func parseSlice(s string) []string {
```

Comment:
would this make sense in the util package so other packages could also use it? it looks general

Reply:
I prefer to leave helper functions like this close to where they're called until there's obvious value in deduplicating them. When they are deduplicated, I prefer to do so in a more targeted package than `util`, which I feel is a mispattern. 

These links provide some context around this:
* https://www.youtube.com/watch?v=PAAkCSZUG1c&t=9m28s
* https://blog.golang.org/package-names
* https://dave.cheney.net/2019/01/08/avoid-package-names-like-base-util-or-common